pkg/api/server/v1alpha2/auth/rbac.go

```diff
@@ -114,3 +155,19 @@ func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) erro
 	// token or unauthorized user, so this is safer to not leak any state.
 	return status.Error(codes.Unauthenticated, "permission denied")
 }
+
+// convertExtra converts the map[string][]string to map[string]ExtraValue for Subject Access Review.
+func convertExtra(extra map[string][]string) map[string]authzv1.ExtraValue {
+	var newExtra map[string]authzv1.ExtraValue
+	for key, value := range extra {
+		newExtra[key] = value
+	}
+	return newExtra
+}
+
+// WithImpersonation is an option function to enable Impersonation
+func WithImpersonation() Option {
```

Comment:
With this kind of builder/factory pattern, you should accept an argument so it can be explicitly set to true or false.

```suggestion
func WithImpersonation(enabled bool) Option {
```

Reply:
Added argument